### ISSUE TITLE
Cannot figure out how to record sound

### ISSUE THREAD

**Issue by brkorb** [MaartenBaert/ssr#469]
There is an "audio input" section and I've tried both ALSA and PulseAudio as the backends.  Neither one will record the sound from streaming done via Chrome.  I can hear the sound just fine as it is recorded, but the recording is silent.  Googling doesn't get me to the magic sauce.  I have no real idea what backend or source I should be using.  I run the flash recording and I see video and hear sound.  Is there a readme/howto/something I should look at?

**Comment by iamgreaser** [comment]
You could run JACK and use the ALSA JACK plugin. It's a bit tricky but that's what I would do in this case.

**Comment by brkorb** [comment]
I don't have any programs that start with the four letters, "jack".  I'm not a multi-media junkie, I just need to record a live class online that I cannot attend and don't otherwise know how to record. :)

**Comment by iamgreaser** [comment]
1. Install qjackctl + jack
2. Install the ALSA JACK plugin
3. Run: pasuspender qjackctl
4. Set JACK up to use 3 buffers 
5. Start JACK
6. Follow the instructions under "Using the ALSA JACK PCM plugin" here: http://jackaudio.org/faq/routing_alsa.html

Note, you'll probably want to dummy out that `.asoundrc` config once you're done.

**Comment by brkorb** [comment]
My platform is openSUSE Leap (42.1).  I installed qjackctl plus all the stuff it depended on.  I told JACK to use 3 buffers and started it.  My video won't run at all now.  It starts up frozen.  I can advance the video by clicking the time bar, but all I get are stills.  Even with stopping JACK.  Quitting "pasuspender qjackctl" allows the video to run.  Should I just start JACK and then quit the qjackctl?  Meaning step 5 ought to be "start JACK and quit qjackctl"?

**Comment by brkorb** [comment]
Oh, I just tried to quit qjackctl with JACK running.  I am not allowed to do that.

**Comment by iamgreaser** [comment]
Did you do the asoundrc step?

You'll also want to make sure SSR is built with JACK support.

**Comment by brkorb** [comment]
> $ cat ~/.asoundrc 
> pcm.rawjack {
>     type jack
>     playback_ports {
>         0 system:playback_1
>         1 system:playback_2
>     }
>     capture_ports {
>         0 system:capture_1
>         1 system:capture_2
>     }
> }
> 
> pcm.jack {
>     type plug
>     slave { pcm "rawjack" }
>     hint {
>         description "JACK Audio Connection Kit"
>     }
> }
> 
> pcm.!default {
>     type plug
>     slave { pcm "rawjack" }
> }

What does "SSR is built with JACK support" mean?  (BTW, thank you _SO_MUCH_ for all your assistance!!)

**Comment by iamgreaser** [comment]
See the dropdown menu that says ALSA or PulseAudio? Click on it and make it say JACK instead.

**Comment by brkorb** [comment]
![jack](https://cloud.githubusercontent.com/assets/1242936/16547799/cbd3eae2-412d-11e6-82cf-129286c82cdb.jpg)
That dropdown menu?

**Comment by iamgreaser** [comment]
No, the SSR one.

**Comment by brkorb** [comment]
Ah.  I can't get that far.  If qjackctl is running, then flash does not run at all.  It freezes.  With SSR running with a connection to JACK, the result is that it freezes.  The SSR log file does not seem to have anything interesting in it, tho I could paste it.  There are no time stamps in the log either.

qjackctl has a more interesting log:

> 14:59:05.129 JACK was started with PID=15655.
> Cannot create thread 1 Operation not permitted
> Cannot create thread 1 Operation not permitted
> Cannot create thread 1 Operation not permitted
> jackdmp 1.9.9.5
> Copyright 2001-2005 Paul Davis and others.
> Copyright 2004-2012 Grame.
> jackdmp comes with ABSOLUTELY NO WARRANTY
> This is free software, and you are welcome to redistribute it
> under certain conditions; see the file COPYING for details
> JACK server starting in realtime mode with priority 10
> Cannot lock down 86605722 byte memory area (Cannot allocate memory)
> audio_reservation_init
> Acquire audio card Audio0
> creating alsa driver ... hw:0|hw:0|1024|3|48000|0|0|nomon|swmeter|-|32bit
> configuring for 48000Hz, period = 1024 frames (21.3 ms), buffer = 3 periods
> ALSA: final selected sample format for capture: 32bit integer little-endian
> ALSA: use 3 periods for capture
> ALSA: final selected sample format for playback: 32bit integer little-endian
> ALSA: use 3 periods for playback
> Cannot use real-time scheduling (RR/10)(1: Operation not permitted)
> AcquireSelfRealTime error
> 14:59:07.344 JACK connection change.
> 14:59:07.365 Server configuration saved to "/u/bkorb/.jackdrc".
> 14:59:07.366 Statistics reset.
> 14:59:07.380 Client activated.
> 14:59:07.381 Patchbay deactivated.
> 14:59:07.409 JACK connection graph change.
> Cannot lock down 86605722 byte memory area (Cannot allocate memory)
> 14:59:23.251 JACK connection graph change.
> 14:59:23.422 JACK connection change.

Unclear why there is a problem with an 86 MB allocation.  I have 64GB of swap, too:

> $ cat /proc/meminfo 
> MemTotal:       32914748 kB
> MemFree:        19285240 kB
> MemAvailable:   29282068 kB
> Buffers:             940 kB
> Cached:         10225528 kB
> [...]

**Comment by iamgreaser** [comment]
Yeah it's a warning pertaining to realtime / memlock stuff, you can usually just ignore it.

Try playing some music through ALSA to make sure your ALSA setup is correct. Do you have the ALSA JACK plugin (it's a plugin for ALSA)?

**Comment by brkorb** [comment]
I have to leave in about 15 minutes.  I really only use two apps for sound:  chrome and skype.  Since I don't have music files, I wouldn't know how to use alsa to run audio recordings.

**Comment by iamgreaser** [comment]
Look through your package manager to see if there's something like alsa-jack or whatever.

Until then, rename that .asoundrc to something else so your sound works in the meantime.
